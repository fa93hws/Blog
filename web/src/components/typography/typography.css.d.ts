// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'base': string;
  'defaultFontStack': string;
  'fontSizeMedium': string;
  'fontSizeMicro': string;
  'fontSizeSmall': string;
  'fontSizeTitleLarge': string;
  'fontSizeTitleMedium': string;
  'fontSizeTitleSmall': string;
  'fontWeightBold': string;
  'fontWeightRegular': string;
  'fontWeightSemiBold': string;
  'textBase': string;
  'textMedium': string;
  'textMediumLineHeight': string;
  'textMicro': string;
  'textMicroLineHeight': string;
  'textSmall': string;
  'textSmallLineHeight': string;
  'textTitleLargeLineHeight': string;
  'textTitleMediumLineHeight': string;
  'textTitleSmallLineHeight': string;
  'titleBase': string;
  'titleLarge': string;
  'titleMedium': string;
  'titleSmall': string;
}
export const cssExports: CssExports;
export default cssExports;
