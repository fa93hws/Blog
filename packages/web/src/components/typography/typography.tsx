import * as React from 'react';
import styles from './typography.css';

type HeadingTagName = keyof Pick<
  JSX.IntrinsicElements,
  'h1' | 'h2' | 'h3' | 'h4' | 'h5'
>;
type TagName = HeadingTagName | keyof Pick<JSX.IntrinsicElements, 'p'>;

function createTextComponent({
  tagName,
  className,
}: {
  tagName: TagName;
  className: string;
}) {
  const TextElement = ({ children }: { children: React.ReactNode }) =>
    React.createElement(tagName, { className }, children);
  return React.memo(TextElement);
}

export const Title = {
  Large: createTextComponent({
    tagName: 'h2',
    className: styles.titleLarge,
  }),
  Medium: createTextComponent({
    tagName: 'h3',
    className: styles.titleMedium,
  }),
  Small: createTextComponent({
    tagName: 'h4',
    className: styles.titleSmall,
  }),
};

export const Text = {
  Medium: createTextComponent({
    tagName: 'p',
    className: styles.textMedium,
  }),
  Small: createTextComponent({
    tagName: 'p',
    className: styles.textSmall,
  }),
  Micro: createTextComponent({
    tagName: 'p',
    className: styles.textMicro,
  }),
};
