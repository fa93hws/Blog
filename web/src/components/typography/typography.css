@import url('../fonts/font.css');
@value defaultFontStack: 'Open Sans', -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif;

@value fontSizeTitleLarge: 3.2rem;
@value fontSizeTitleMedium: 2.4rem;
@value fontSizeTitleSmall: 1.8rem;
@value fontSizeMedium: 1.4rem;
@value fontSizeSmall: 1.2rem;
@value fontSizeMicro: 1.0rem;

@value fontWeightRegular: 400;
@value fontWeightSemiBold: 600;
@value fontWeightBold: 700;

@value textTitleLargeLineHeight: 1.3;
@value textTitleMediumLineHeight: 1.3;
@value textTitleSmallLineHeight: 1.3;
@value textMediumLineHeight: 1.6;
@value textSmallLineHeight: 1.6;
@value textMicroLineHeight: 1.6;

.base {
  font-family: defaultFontStack;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-smooth: always;
}

.titleBase {
  composes: base;
  margin: 0 auto;
}

.textBase {
  composes: base;
}

.titleLarge {
  composes: titleBase;

  font-size: fontSizeTitleLarge;
  font-weight: fontWeightBold;
  letter-spacing: -0.05rem;
  line-height: textTitleLargeLineHeight;
}

.titleMedium {
  composes: titleBase;

  font-size: fontSizeTitleMedium;
  letter-spacing: -0.03rem;
  line-height: textTitleMediumLineHeight;
}

.titleSmall {
  composes: titleBase;

  font-size: fontSizeTitleSmall;
  letter-spacing: -0.01rem;
  line-height: textTitleSmallLineHeight;
}

.textMedium {
  composes: textBase;

  font-weight: fontWeightRegular;
  font-size: fontSizeMedium;
  line-height: textMediumLineHeight;
}

.textSmall {
  composes: textBase;

  font-weight: fontWeightRegular;
  font-size: fontSizeSmall;
  line-height: textSmallLineHeight;
}

.textMicro {
  composes: textBase;

  font-weight: fontWeightSemiBold;
  font-size: fontSizeMicro;
  line-height: textMicroLineHeight;
}
