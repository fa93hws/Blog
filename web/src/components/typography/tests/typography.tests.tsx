import * as React from 'react';
import { Title, Text } from '../typography';

describe('Title', () => {
  it('renders large title', () => {
    expect(<Title.Large>Large</Title.Large>).toMatchRenderedSnapshot();
  });

  it('renders medium title', () => {
    expect(<Title.Medium>Medium</Title.Medium>).toMatchRenderedSnapshot();
  });

  it('renders small title', () => {
    expect(<Title.Small>Small</Title.Small>).toMatchRenderedSnapshot();
  });
});

describe('Text', () => {
  it('renders medium text', () => {
    expect(<Text.Medium>Medium</Text.Medium>).toMatchRenderedSnapshot();
  });

  it('renders small text', () => {
    expect(<Text.Small>Medium</Text.Small>).toMatchRenderedSnapshot();
  });

  it('renders micro text', () => {
    expect(<Text.Micro>Micro</Text.Micro>).toMatchRenderedSnapshot();
  });
});
