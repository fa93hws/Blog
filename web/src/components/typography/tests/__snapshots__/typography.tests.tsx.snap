// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders medium text 1`] = `
<p
  class="textMedium"
>
  Medium
</p>
`;

exports[`Text renders micro text 1`] = `
<p
  class="textMicro"
>
  Micro
</p>
`;

exports[`Text renders small text 1`] = `
<p
  class="textSmall"
>
  Medium
</p>
`;

exports[`Title renders large title 1`] = `
<h2
  class="titleLarge"
>
  Large
</h2>
`;

exports[`Title renders medium title 1`] = `
<h3
  class="titleMedium"
>
  Medium
</h3>
`;

exports[`Title renders small title 1`] = `
<h4
  class="titleSmall"
>
  Small
</h4>
`;
